File: bob/bob.exs
defmodule Teenager do

  def hey(what) do
    cond do
      nil? String.first what ->
        "Fine. Be that way."
      what == String.upcase(what) ->
        "Woah, chill out!"
      String.ends_with? what, "?" ->
        "Sure."
      true ->
        "Whatever."
    end
  end

end

File: haskell/bob/bob.hs
module Bob (responseFor) where
import Data.Char

isSilence :: String -> Bool
isSilence sentence = all isSpace sentence

isYelling :: String -> Bool
isYelling sentence = any isAlpha sentence && all isUpper (filter isAlpha sentence)

isQuestion :: String -> Bool
isQuestion sentence = last sentence == '?'

responseFor :: String -> String
responseFor sentence
        | isSilence sentence  = "Fine. Be that way!"
        | isYelling sentence  = "Woah, chill out!"
        | isQuestion sentence = "Sure."
        | otherwise           = "Whatever."
